skbeam/core/correlation.py

```diff
@@ -842,3 +850,322 @@ def one_time_from_two_time(two_time_corr):
         for j in range(two_time_corr.shape[2]):
             one_time_corr[:, j] = np.trace(g, offset=j)/two_time_corr.shape[2]
     return one_time_corr
+
+
+class CrossCorrelator:
+    '''
+        Compute a 1D or 2D cross-correlation on data.
+
+        This uses a mask, which may be binary (array of 0's and 1's),
+        or a list of non-negative integer id's to compute cross-correlations
+        separately on.
+
+        The symmetric averaging scheme introduced here is inspired by a paper
+        from Schätzel, although the implementation is novel in that it allows
+        for the usage of arbitrary masks. Reference to Schatzel's original
+        paper is here:
+            Schätzel, Klaus, Martin Drewel, and Sven Stimac. “Photon
+            correlation measurements at large lag times: improving statistical
+            accuracy.” Journal of Modern Optics 35.4 (1988): 711-718.
```

Comment:
thanks for adding this.  You may modify it a little bit like https://github.com/scikit-beam/scikit-beam/blob/master/skbeam/core/fitting/background.py#L114  Or check numpy docstrings.